content/zh/docs/faq/_index.md

```diff
@@ -0,0 +1,408 @@
+身份和授权特性。Helm的权限通过你的
+[kubeconfig文件](https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/)进行评估。
+集群管理员可以限制用户权限，只要他们觉得合适，
+无论什么粒度都可以做到。版本发布记录和Helm的剩余保留功能仍然会被记录在集群中。
+
+### 改进升级策略: 三路策略合并补丁
+
+Helm 2 使用了一种双路策略合并补丁。在升级过程中，会对比最近一次的chart manifest和提出的
+chart manifest(通过`helm upgrade`提供)。升级会对比两个chart的不同来决定哪些更改会应用到Kubernetes资源中。
+如果更改是集群外带的（比如通过`kubectl edit`），则不会被考虑。结果就是资源不会回滚到之前的状态：
+因为Helm只考虑最后一次应用的chart manifest作为它的当前状态，如果chart状态没有更改，则资源的活动状态不会更改。
+
+现在Helm 3中，我们使用一种三路策略来合并补丁。Helm在生成一个补丁时会考虑之前老的manifest的活动状态。
+
+#### 示例
+
+让我们通过一些常见的例子来看看变化带来的影响。
+
+##### 回滚已经改变的活动状态
+
+你的团队正好在Kubernetes上使用Helm部署了生产环境应用。chart包含了一个部署对象使用了三套副本：
+
+```console
+$ helm install myapp ./myapp
+```
+
+&emsp;&emsp;一个开发新人加入了团队。当他们第一点观察生产环境集群时，发生了一个像是咖啡洒在了键盘上一样的严重事故，
+他们使用 `kubectl scale` 对生产环境部署进行缩容，将副本数从3降到了0 。
+
+```console
+$ kubectl scale --replicas=0 deployment/myapp
+```
+
+&emsp;&emsp;团队里面的另一个人看到线上环境已经挂了就决定回滚这个版本到之前的状态：
+
+```console
+$ helm rollback myapp
+```
+
+发生了什么？
+
+在Helm 2中，会生成一个补丁并对比老的manifest和新的manifest。因为这是一个回滚，manifest是一样的。
+Helm会认为新老manifest没有区别，因此没有需要更改的内容。副本统计数量继续保持为0。恐慌就接踵而至。
+
+在Helm 3中，是用老的manifest生成新的补丁，活动状态和新的manifest。
+Helm 意识到老的状态是3，而现有活动状态是0，并且新的manifest希望改回3，因此会生成一个补丁将状态改回3。
+
+##### 活动状态已更改的情况下升级
+
+很多服务网格和其他基于controller的应用向Kubernetes对象中注入数据。比如sidecar、label和其他信息。
+之前如果你从Chart渲染给定的manifest如下:
+
+```yaml
+containers:
+- name: server
+  image: nginx:2.0.0
+```
+
+并且另一个应用修改活动状态如下：
+
+```yaml
+containers:
+- name: server
+  image: nginx:2.0.0
+- name: my-injected-sidecar
+  image: my-cool-mesh:1.0.0
+```
+
+现在你想升级`nginx`镜像到`2.1.0`。因此用指定的manifest升级chart：
+
+```yaml
+containers:
+- name: server
+  image: nginx:2.1.0
+```
+
+发生了什么？
+
+在Helm 2中，Helm 在新老manifest之间生成了一个`containers`对象的补丁。
+生成补丁的过程中不考虑集群的活动状态。
+
+集群的活动状态被修改成了这样:
+
+```yaml
+containers:
+- name: server
+  image: nginx:2.1.0
+```
+
+sidecar pod从活动状态中移除了。更多的恐慌袭来。
+
+在Helm 3中，Helm 在新的manifest、活动状态和老manifest之间生成了一个`containers`对象的补丁。
+会注意到新的manifest将镜像tag更新为`2.1.0`, 但是活动状态中包含了一个sidecar容器。
+
+集群的活动状态被修改成了下面这样：
+
+```yaml
+containers:
+- name: server
+  image: nginx:2.1.0
+- name: my-injected-sidecar
+  image: my-cool-mesh:1.0.0
+```
+
+### 发布名称现在限制在namespace范围内
+
+&emsp;&emsp;随着Tiller的移除， 每个版本的信息需要保存在某个地方。
+在Helm 2中，是存储在Tiller相同的命名空间中。 
+实际上这意味着一个发布版本使用一个名称，其他发布不能使用相同的名称，
+即使在不同的命名空间中也不行。
+
+&emsp;&emsp;在Helm 3中，特定的版本信息作为发布本身存储在相同的命名空间中。
+意味着用户现在可以在两个分开的命名空间中使用`helm install wordpress stable/wordpress`，
+并且每个都能使用 `helm list` 改变当前命名空间。 (例如 `helm list --namespace foo`)。
+
+&emsp;&emsp;与本地集群命名空间更好的一致性，使得 `helm list` 命令不再需要默认列出所有发布版本的列表。
+取而代之的是，仅仅会在命名空间中列出当前kubernetes上下文的版本。
+(也就是说运行`kubectl config view --minify`时会显示命名空间). 也就意味着您在执行`helm list`时必须提供
+ `--all-namespaces` 标识才能获得和Helm 2同样的结果。
+
+### 作为默认存储器的密钥
+
+In Helm 3, 密钥被作为[默认存储驱动](https://helm.sh/docs/topics/advanced/#storage-backends)使用。
+Helm 2默认使用ConfigMaps记录版本信息。在Helm 2.7.0中，新的存储后台使用密钥来存储版本信息，
+现在是Helm 3的默认设置。
+
+Helm 3默认允许更改密钥作为额外的安全措施在Kubernetes中和密钥加密一起保护chart。
```

Comment:
> Changing to Secrets as the Helm 3 default allows for additional security in protecting charts in conjunction with the release of Secret encryption in Kubernetes.

Does your translation say that Helm 3 uses Secrets as the default (instead of using the previous default which was ConfigMaps)?  Thanks.

Reply:
Yes，I confirmed.